code/datums/components/anti_magic.dm

```diff
@@ -1,12 +1,14 @@
 /datum/component/anti_magic
 	var/magic = FALSE
 	var/holy = FALSE
+	var/tinfoil = FALSE
+	var/checkslot
 	var/charges = INFINITY
 	var/blocks_self = TRUE
 	var/datum/callback/reaction
 	var/datum/callback/expire
 
-/datum/component/anti_magic/Initialize(_magic = FALSE, _holy = FALSE, _charges, _blocks_self = TRUE, datum/callback/_reaction, datum/callback/_expire)
+/datum/component/anti_magic/Initialize(_magic = FALSE, _holy = FALSE, _tinfoil = FALSE, _checkslot, _charges, _blocks_self = TRUE, datum/callback/_reaction, datum/callback/_expire)
```

Comment:
tinfoil is not a kind of power, name it more according to what it is like psychic or something